src/NextGen/Gateways/PayPalCommerce/PayPalCommerceSubscriptionModule.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Give\NextGen\Gateways\PayPalCommerce;
+
+use Exception;
+use Give\Donations\Models\Donation;
+use Give\Framework\PaymentGateways\Commands\GatewayCommand;
+use Give\Framework\PaymentGateways\Commands\SubscriptionProcessing;
+use Give\Framework\PaymentGateways\SubscriptionModule;
+use Give\Subscriptions\Models\Subscription;
+
+class PayPalCommerceSubscriptionModule extends SubscriptionModule
+{
+    /**
+     * @unreleased
+     */
+    public function createSubscription(
+        Donation $donation,
+        Subscription $subscription,
+        $gatewayData = null
+    ): GatewayCommand {
+
+        $subscriptionId = $gatewayData['payPalSubscriptionId'];
+
+        return new SubscriptionProcessing($subscriptionId);
+    }
+
+    public function cancelSubscription(Subscription $subscription)
+    {
+        // TODO: Implement cancelSubscription() method.
+    }
```

Comment:
@kjohnson You will need to add this logic for cancelling a subscription like in the [PayPal standard module](https://github.com/impress-org/givewp-next-gen/blob/6ccb343f08ab5e0d8fa52dbdfb5ca56ef5e76138/src/NextGen/Gateways/PayPal/PayPalStandardGateway/PayPalStandardGatewaySubscriptionModule.php#L205).

You can test this out in the donor dashboard by clicking cancel subscription on an active subscription and make sure it actually gets cancelled in PayPal.

This should be the last piece to this integration as everything else appears to be working properly.

Reply:
Ah, the donor dashboard. I had tested via the Subscription details screen, which already works with the gateway.

Reply:
## Call Summary

The PayPal Commerce recurring gateway was hooked in to the GiveWP Subscription status change and synced that new status with PayPal.

Due to changes in the Recurring add-on, this cancellation needed to be defined in the Subscription Module.

The solution was to call the PayPal Commerce Subscription Repository `updateStatus()` method from the subscription module cancel method.